### ISSUE TITLE
screensharing not working on hyprland, but works on sway

### ISSUE THREAD

**Issue by x0rzavi** [hyprwm/xdg-desktop-portal-hyprland#46]
Screen sharing isn't working on neither obs nor firefox, it just shows up as a blank region. However everything is fine on sway. I've tried to reproduce the workflow as much as possible on both the WMs.

`hyprland: v0.24.1`
`xdg-desktop-portal-hyprland: v0.2.1`

I'm on gentoo and both the packages are from `::guru` repository. I've attached the corresponding logs for pipewire, obs and xdph on sway and hyprland below.

[obs_hyprland.txt](https://github.com/hyprwm/xdg-desktop-portal-hyprland/files/11234960/obs_hyprland.txt)
[obs_sway.txt](https://github.com/hyprwm/xdg-desktop-portal-hyprland/files/11234961/obs_sway.txt)

[pipewire_hyprland.txt](https://github.com/hyprwm/xdg-desktop-portal-hyprland/files/11234962/pipewire_hyprland.txt)
[pipewire_sway.txt](https://github.com/hyprwm/xdg-desktop-portal-hyprland/files/11234963/pipewire_sway.txt)

[xdph_hyprland.txt](https://github.com/hyprwm/xdg-desktop-portal-hyprland/files/11234964/xdph_hyprland.txt)
[xdph_sway.txt](https://github.com/hyprwm/xdg-desktop-portal-hyprland/files/11234965/xdph_sway.txt)

**Comment by vaxerski** [comment]
nvidia?

**Comment by x0rzavi** [comment]
tried that, unfortunately it doesn't fix anything. Guess I'll just wait for a future release to magically fix the issue :)

**Comment by IDontKnow2Code** [comment]
grim also says `failed to copy output eDP-1`

**Comment by x0rzavi** [comment]
> grim also says `failed to copy output eDP-1`

can confirm the same on v0.24.1

**Comment by vaxerski** [comment]
brilliant, maybe instead of adding noise you consider adding the thing I asked for to debug this issue further?

**Comment by vaxerski** [comment]
is your system little endian?

**Comment by x0rzavi** [comment]
> is your system little endian?

idk, I use a fairly normal system ig? on a Ryzen 5 3500u.

Here are the hyprland logs while trying to screenshare 
[hyprland.txt](https://github.com/hyprwm/xdg-desktop-portal-hyprland/files/11251410/hyprland.txt)

**Comment by vaxerski** [comment]
check on -git with 25f14294a84df1a2a840a92fd89ff4e1fd72f01e

**Comment by x0rzavi** [comment]
Screensharing still fails.
Screenshots no longer fail but produces a black image, logs below

```
[LOG] LayerSurface 7836b130 arranged: x: 0 y: 1034 w: 1920 h: 46 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] LayerSurface 784d1ff0 arranged: x: 0 y: 0 w: 1920 h: 1080 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] Monitor eDP-1 layers arranged: reserved: 0.000000 0.000000 0.000000 46.000000
[LOG] Cleanup: destroyed a layersurface
[LOG] Hiding the cursor (timeout)
[LOG] Keybind triggered, calling dispatcher (64, , 65377)
[LOG] Executing WAYLAND_DISPLAY=wayland-1 DISPLAY=:0 grimblast --notify copysave output ~/Pictures/Screenshots/$(date +%F+%H:%M:%S).png
[LOG] Process Created with pid 3695
[LOG] ScreencopyProtocolManager bound successfully!
[LOG] New LayerSurface has no preferred monitor. Assigning Monitor eDP-1
[LOG] Registered signal for owner 7886b190: 78861708 -> 7886b368 (owner: layerSurface)
[LOG] Registered signal for owner 7886b190: 77ab1c00 -> 7886b230 (owner: layerSurface)
[LOG] Registered signal for owner 7886b190: 77ab1c10 -> 7886b298 (owner: layerSurface)
[LOG] Registered signal for owner 7886b190: 77ab1c20 -> 7886b300 (owner: layerSurface)
[LOG] Registered signal for owner 7886b190: 77ab1c30 -> 7886b3d0 (owner: layerSurface)
[LOG] LayerSurface 77ab1b40 (namespace notifications layer 2) created on monitor eDP-1
[LOG] LayerSurface 7836b130 arranged: x: 0 y: 1034 w: 1920 h: 46 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] LayerSurface 784d1ff0 arranged: x: 0 y: 0 w: 1920 h: 1080 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] LayerSurface 7836b138 arranged: x: 1620 y: 934 w: 300 h: 100 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] Monitor eDP-1 layers arranged: reserved: 0.000000 0.000000 0.000000 46.000000
[LOG] LayerSurface 7836b130 arranged: x: 0 y: 1034 w: 1920 h: 46 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] LayerSurface 784d1ff0 arranged: x: 0 y: 0 w: 1920 h: 1080 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] LayerSurface 7836b138 arranged: x: 1620 y: 826 w: 300 h: 208 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] Monitor eDP-1 layers arranged: reserved: 0.000000 0.000000 0.000000 46.000000
[LOG] LayerSurface 77ab1b40 mapped
[LOG] Registered signal for owner 7886b1a8: 78861728 -> 7886b1b0 (owner: CWLSurface)
[LOG] CWLSurface 7886b1a8 called init()
[LOG] LayerSurface 7836b130 arranged: x: 0 y: 1034 w: 1920 h: 46 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] LayerSurface 784d1ff0 arranged: x: 0 y: 0 w: 1920 h: 1080 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] LayerSurface 7836b138 arranged: x: 1620 y: 826 w: 300 h: 208 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] Monitor eDP-1 layers arranged: reserved: 0.000000 0.000000 0.000000 46.000000
```

**Comment by vaxerski** [comment]
nvidia? if so, black screen means no nvidia patches

**Comment by x0rzavi** [comment]
amd igpu :)

**Comment by milkknife** [comment]
> Try using -git versions of both

this fixed it for me (AUR)

**Comment by IDontKnow2Code** [comment]
> nvidia? if so, black screen means no nvidia patches

Mine Intel iris xe graphics, it gives black output.
Log below

```
[LOG] ScreencopyProtocolManager bound successfully!
[LOG] Registered signal for owner 55e0706f1db0: 55e071fbfe18 -> 55e0706f1f88 (owner: layerSurface)
[LOG] Registered signal for owner 55e0706f1db0: 55e071fc0080 -> 55e0706f1e50 (owner: layerSurface)
[LOG] Registered signal for owner 55e0706f1db0: 55e071fc0090 -> 55e0706f1eb8 (owner: layerSurface)
[LOG] Registered signal for owner 55e0706f1db0: 55e071fc00a0 -> 55e0706f1f20 (owner: layerSurface)
[LOG] Registered signal for owner 55e0706f1db0: 55e071fc00b0 -> 55e0706f1ff0 (owner: layerSurface)
[LOG] LayerSurface 55e071fbffc0 (namespace notifications layer 3) created on monitor eDP-1
[LOG] LayerSurface 55e071fa0b20 arranged: x: 0 y: 0 w: 1920 h: 45 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] LayerSurface 55e07171fb10 arranged: x: 0 y: 0 w: 1920 h: 1080 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] LayerSurface 55e071fbb300 arranged: x: 1409 y: 59 w: 497 h: 88 with margins: t: 14 l: 14 r: 14 b: 14
[LOG] Monitor eDP-1 layers arranged: reserved: 0.000000 45.000000 0.000000 0.000000
[LOG] LayerSurface 55e071fbffc0 mapped
[LOG] Registered signal for owner 55e0706f1dc8: 55e071fbfe38 -> 55e0706f1dd0 (owner: CWLSurface)
[LOG] CWLSurface 55e0706f1dc8 called init()
[LOG] LayerSurface 55e071fa0b20 arranged: x: 0 y: 0 w: 1920 h: 45 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] LayerSurface 55e07171fb10 arranged: x: 0 y: 0 w: 1920 h: 1080 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] LayerSurface 55e071fbb300 arranged: x: 1409 y: 59 w: 497 h: 88 with margins: t: 14 l: 14 r: 14 b: 14
[LOG] Monitor eDP-1 layers arranged: reserved: 0.000000 45.000000 0.000000 0.000000
[LOG] LayerSurface 55e071fbffc0 unmapped
[LOG]  [hookSystem] New hook event registered: closeLayer
[LOG] Framebuffer created, status 36053
[LOG] Callback 55e0706f1df8 -> 55e0706f1df0, CWLSurface removed.
[LOG] CWLSurface 55e0706f1dc8 called destroy()
[LOG] LayerSurface 55e071fbffc0 destroyed
[LOG] Callback 55e0706f1fb0 -> 55e0706f1fa8, layerSurface removed.
[LOG] Callback 55e0706f1e78 -> 55e0706f1e70, layerSurface removed.
[LOG] Callback 55e0706f1ee0 -> 55e0706f1ed8, layerSurface removed.
[LOG] Callback 55e0706f1f48 -> 55e0706f1f40, layerSurface removed.
[LOG] Callback 55e0706f2018 -> 55e0706f2010, layerSurface removed.
[LOG] LayerSurface 55e071fa0b20 arranged: x: 0 y: 0 w: 1920 h: 45 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] LayerSurface 55e07171fb10 arranged: x: 0 y: 0 w: 1920 h: 1080 with margins: t: 0 l: 0 r: 0 b: 0
[LOG] Monitor eDP-1 layers arranged: reserved: 0.000000 45.000000 0.000000 0.000000
[LOG] Cleanup: destroyed a layersurface

```

**Comment by IDontKnow2Code** [comment]
> nvidia? if so, black screen means no nvidia patches

I have Intel iris xe graphics no nvidia.
It still gives me black output.
I use `hyprland-git` and `grim-git` from AUR.

**Comment by mordax7** [comment]
For me the screen sharing works on Firefox.

Chromium broke since the upgrade from version `111` to `112`.

**Comment by devadathanmb** [comment]
> For me the screen sharing works on Firefox.
> 
> Chromium broke since the upgrade from version `111` to `112`.

Yes I also have the same issue. It works on Firefox but not on Chromium 112. Reverting back to Chromium 111 fixes the issue. 

Also I've tested this on Sway with xdg-desktop-portal-wlr 0.7.0-1 and it works fine there. So seems like a hyprland specific issue?

**Comment by vaxerski** [comment]
I was pretty sure I fixed it, maybe I broke it again. Ugh.

**Comment by vaxerski** [comment]
fixed in c0e233955568fbea4e859336f6d3d14d51294d7c

**Comment by x0rzavi** [comment]
> fixed in [c0e2339](https://github.com/hyprwm/xdg-desktop-portal-hyprland/commit/c0e233955568fbea4e859336f6d3d14d51294d7c)

Unfortunately, neither did it fix my screensharing nor did it fix the black screenshot issue others have reported too :)

**Comment by devadathanmb** [comment]
> > fixed in [c0e2339](https://github.com/hyprwm/xdg-desktop-portal-hyprland/commit/c0e233955568fbea4e859336f6d3d14d51294d7c)
> 
> Unfortunately, neither did it fix my screensharing nor did it fix the black screenshot issue others have reported too :)
> 
> Edit: I've compiled both `hyprland` and `xdg-desktop-portal-hyprland` from the latest commits at the time of posting of this comment

I can confirm too. This bug was fixed at some point as pointed out in this thread but seems like it has appeared again.

**Comment by devadathanmb** [comment]
> I have mistakenly turned on bitdepth 10, which is definitely the root cause here in my case. Everything seems fine now, both screensharing and screenshot work as usual.

Although you had something else this actually was an issue with hyprland lol. Anyways, seems like it has been fixed.
